### PR TITLE
Temporary removing Microsoft.VisualBasic.Tests.csproj from the tests builds

### PR DESCRIPTION
**Pull request by joperezr**
Because of Microsoft/MSBuild#137 Microsoft.VisualBasic.vbproj can't be built in Unix so it has been disabled for some time. Now we have a build break because even though the product side project build is being disabled, the test build is trying to build the project again, causing the build to fail. This change temporarily disables the VisualBasic test project from building.

cc: @weshaggard @sokket

### CODE REVIEW

**Comment by joperezr** [comment]
@weshaggard Looks like the CI builds on PRs are currently broken so @sokket cannot merge his change which would fix it. Is it ok to merge this (smaller and simpler workaround) fix without triggering a CI given that the workflow (calling build.sh) in non-Windows platforms is broken now?

If it is ok with you guys, I would like to unblock the non-Windows builds and then remove this workaround once Jon is ready to merge his stuff.

**Comment by joperezr** [comment]
Also, I did run a full build (called build.sh) on my OS X machine and the build after this change seems to be all green again.

**Comment by mellinoe** [comment]
Yeah I see this as well. I'm not sure how this one wasn't getting hit before (when i disabled the source project initially), but this looks like an okay fix for now.

File: src/tests.builds [hunk 1]
```diff
@@ -2,12 +2,17 @@
...
-    <Project Include="*\test*\**\*.csproj">
+    <Project Include="*\test*\**\*.csproj" Condition="'$(OSEnvironment)'=='Windows_NT'">
       <OSGroup Condition="'$(FilterToOSGroup)'!=''">$(FilterToOSGroup)</OSGroup>
     </Project>
     <Project Include="*\test*\**\*.vbproj" Condition="'$(IncludeVbProjects)'!='false'">
       <OSGroup Condition="'$(FilterToOSGroup)'!=''">$(FilterToOSGroup)</OSGroup>
     </Project>
+
+    <!-- Temp Workaround to unblock non-windows builds -->
+    <Project Include="*\test*\**\*.csproj" Exclude="**\Microsoft.VisualBasic.Tests.csproj" Condition="'$(OSEnvironment)'!='Windows_NT'">
```
Comment (weshaggard) [inline src/tests.builds]: No need to separate these like this. Instead just create an ExcludeProjects itemgroup and put this test project in it based on the OS condition.

File: src/tests.builds [hunk 2]
```diff
@@ -2,12 +2,17 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
-    <Project Include="*\test*\**\*.csproj">
+    <Project Include="*\test*\**\*.csproj" Condition="'$(OSEnvironment)'=='Windows_NT'">
       <OSGroup Condition="'$(FilterToOSGroup)'!=''">$(FilterToOSGroup)</OSGroup>
     </Project>
     <Project Include="*\test*\**\*.vbproj" Condition="'$(IncludeVbProjects)'!='false'">
```
Comment (weshaggard) [inline src/tests.builds]: I'm not sure where IncludeVbProjects gets set but perhaps that is the condition you should use here as well.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by joperezr** [comment]
@dotnet-bot test this please

File: src/tests.builds (hunk 1 above)
Comment (joperezr) [inline src/tests.builds]: Great idea. I just pushed a change with that suggestion.

**Comment by joperezr** [comment]
cc: @ianhays @dagood

File: BuildToolsVersion.txt [hunk 3]
```diff
@@ -1 +1,2 @@
-1.0.25-prerelease-00167
+<<<<<<< 482e123738f0ceab4a16c145ba57f357c271971d
```
Comment (weshaggard) [inline BuildToolsVersion.txt]: Bad merge?

**Comment by weshaggard** [comment]
Fix the bad merge and squash. Otherwise the change looks good to me.

**Comment by mellinoe** [comment]
LGTM

**Comment by ianhays** [comment]
The buildtools update includes changes to the Content binplacing that will cause errors in System.Reflection.TypeExtensions.CoreCLR.Tests as well as System.Net.NetworkInformation.FunctionalTests.

Those errors are expected and I've got an accompanying PR ready for whenever this merges that will resolve those failures.

**Comment by mellinoe** [comment]
> Those errors are expected and I've got an accompanying PR ready for whenever this merges that will resolve those failures.

Should we try to merge that change in first, or perhaps see if @joperezr can cherry-pick your commit to include those fixes here?

**Comment by joperezr** [comment]
@mellinoe I'll cherry-pick his change

**Comment by ianhays** [comment]
LGTM

**Comment by joperezr** [comment]
Merging to unblock Linux build.

**Comment by stephentoub** [comment]
@joperezr, is there an issue tracking re-enabling this?

**Pull request merged**